pkg/archivetool/archivetool.go

```diff
@@ -12,7 +13,55 @@ import (
 	"strings"
 )
 
-func readArchive(archiveURL string) ([]byte, error) {
+func ArchiveToLocalPath(uri string) (string, func(), error) {
+	b, err := ReadArchive(uri)
+	if err != nil {
+		return "", nil, err
+	}
+
+	// Extract the ZIP archive in a temporary directory.
+	archiveDir, cleanup, err := ExtractPlugin(bytes.NewReader(b))
+	if err != nil {
+		return "", nil, err
```

Comment:
Ops, missed this during the first review, `cleanup` should be called here as well:

```suggestion
		cleanup()
		return "", nil, err
```

otherwise we lose it (+ the caller should not care about it if we return an error)